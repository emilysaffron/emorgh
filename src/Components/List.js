import BulletedList, { BulletedListItem } from "@bbc/psammead-bulleted-list";
import Grid from "@bbc/psammead-grid";
import getGridGroups from "../helpers/GetGridGroups";

const List = ({ items }) => {
  const list = items.map((listItem) => {
    return <BulletedListItem>{listItem}</BulletedListItem>;
  });

  return (
    <Grid
      item
      startOffset={getGridGroups([2, 2, 2, 2, 2, 2])}
      columns={getGridGroups([8, 8, 8, 8, 8, 8])}
    >
      <BulletedList>{list}</BulletedList>
    </Grid>
  );
};
export default List;
