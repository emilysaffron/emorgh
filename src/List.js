import BulletedList, { BulletedListItem } from "@bbc/psammead-bulleted-list";
import Grid from "@bbc/psammead-grid";

const List = ({ items }) => {
  let listLength = items.length;
  let list = [];
  for (let i = 0; i < listLength; i++) {
    list[i] = <BulletedListItem>{items[i]}</BulletedListItem>;
  }

  return (
    <Grid
      item
      startOffset={{
        group0: 2,
        group1: 2,
        group2: 2,
        group3: 2,
        group4: 2,
        group5: 2,
      }}
      columns={{
        group0: 8,
        group1: 8,
        group2: 8,
        group3: 8,
        group4: 8,
        group5: 8,
      }}
    >
      <BulletedList>{list}</BulletedList>
    </Grid>
  );
};
export default List;
