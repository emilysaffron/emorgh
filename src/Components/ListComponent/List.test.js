import List from "./List";
import renderer from "react-test-renderer";

it("List Snapshot", () => {
    const tree = renderer.create(<List items="[]" />).toJSON();
    expect(tree).toMatchSnapshot();
  });