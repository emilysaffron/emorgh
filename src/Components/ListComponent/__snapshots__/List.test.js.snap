// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshot 1`] = `
<div
  className="css-vxrtof-GridComponent e57qer20"
  dir="ltr"
>
  <ul
    className="css-l31ibh-BulletedList e1drcs2w0"
    dir="ltr"
    role="list"
  >
    <li
      className="css-chb3zw-BulletedListItem e1drcs2w1"
      role="listitem"
    >
      [
    </li>
    <li
      className="css-chb3zw-BulletedListItem e1drcs2w1"
      role="listitem"
    >
      ]
    </li>
  </ul>
</div>
`;
